include/SFML/Window/Keyboard.hpp

```diff
@@ -164,11 +321,72 @@ class SFML_WINDOW_API Keyboard
     static bool isKeyPressed(Key key);
 
     ////////////////////////////////////////////////////////////
+    /// \brief Check if a key is pressed
+    ///
+    /// \param code Scancode to check
+    ///
+    /// \return True if the physical key is pressed, false otherwise
+    ///
+    ////////////////////////////////////////////////////////////
+    static bool isKeyPressed(Scancode code);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Localize a physical key to a logical one
+    ///
+    /// \param code Scancode to localize
+    ///
+    /// \return The key corresponding to the scancode under the current
+    ///         keyboard layout used by the operating system, or
+    ///         sf::Keyboard::Unknown when the scancode cannot be mapped
+    ///         to a Key.
+    ///
+    /// \see unlocalize
+    ///
+    ////////////////////////////////////////////////////////////
+    static Key localize(Scancode code);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Identify the physical key corresponding to a logical one
+    ///
+    /// \param key Key to "unlocalize"
+    ///
+    /// \return The scancode corresponding to the key under the current
+    ///         keyboard layout used by the operating system, or
+    ///         sf::Keyboard::sUnknown when the key cannot be mapped
+    ///         to a Keyboard::Scancode.
+    ///
+    /// \see localize
+    ///
+    ////////////////////////////////////////////////////////////
+    static Scancode unlocalize(Key key);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Provide a string representation for a given scancode
+    ///
+    /// From a high level point of view, this conversion corresponds
+    /// somewhat to the string available through sf::Event::TextEvent
+    /// when the given physical key is pressed by the user, when no
+    /// modifiers are involved.
+    ///
+    /// \warning The result is OS-dependent: for example, sf::Keyboard::sLSystem
+    ///          is "Left Meta" on Linux, "Left Windows" on Windows and
+    ///          "Left Command" on macOS.
+    ///
+    /// The current keyboard layout set by the operating system is used to
+    /// interpret the scancode: for example, sf::Keyboard::SemiColon is
+    /// mapped to ";" for layout and to "é" for others.
+    ///
+    /// \return The localized description of the code
+    ///
+    ////////////////////////////////////////////////////////////
+    static String localizedRepresentation(Scancode code);
```

Comment:
I think *description* is better than *name* or *label* because the values are meant to be displayed, not used as something unique like a key in a `map`. With *name* or *label*, I think the idea of the values being constant is conveyed too strongly. Two calls to this functions with the same argument can yield two different strings if the keyboard layout has changed.

> And is there a reasoning behind taking a scan code and not a key code, in this function?

Yes: a key code would always yield the same string. The mapping would be no different than what [Thor's doing](https://github.com/Bromeon/Thor/blob/master/src/InputNames.cpp#L88). And I don't think SFML needs this function too.